src/javascripts/transifex-api/resource.js

```diff
@@ -195,6 +198,31 @@ var resource = module.exports = {
 
       return arr;
     },
+    txUpsertBatchResources: function(category, objects) {
+      // Save all resource objects 
+      resource.batch = objects;
+      resource.category = category;
+      // Start upserting resources
+      this.txUpsertResourceNext();
+    },
+    txUpsertResourceNext: function() {
```

Comment:
Wouldn't it be better if we uploaded a few resources at the time instead of one? This way we would still get no timeouts but have a faster uploading time overall.